lib/find.coffee

```diff
@@ -115,6 +115,16 @@ module.exports =
       'find-and-replace:select-skip': (event) ->
         selectNextObjectForEditorElement(this).skipCurrentSelection()
 
+    # Request autocompletions for the find field.
+    autocompletePackage = atom.packages.activePackages['autocomplete-plus']
+    if autocompletePackage?
+      autocompleteManager = autocompletePackage.mainModule.autocompleteManager
+      for providerData in autocompleteManager.providerManager.providers
+        if providerData.provider.constructor.name is 'SymbolProvider'
```

Comment:
Seems like this could be problematic if the autocomplete-plus package gets activated *after* the find-and-replace package. It also seems like `SymbolProvider` gets re-initialized every time the user toggles the `autocomplete-plus.enableBuiltinProvider`, thus disabling autocompletion until the next reload.

I wonder if we could provide a service in ac+ instead and let it expose an interface like the following:

```coffee
{
  getDefaultProvider: ->
    @defaultProvider

  observeProviders: (callback) ->
    callback(provider) for provider in @providers
    @emitter.on 'did-register-provider', callback
    
  onDidRemoveProvider: (callback) -> 
    @emitter.on 'did-remove-provider', callback
}
```

We might add even more methods to it, but I am a little unclear on what are the expansion points we may want to provide, so I think it's okay to keep the API tight so that we can support just this use case for now, while still potentially allowing package authors to use it too. After we do that, we could use the service like this:

```coffee
consumeAutocompletePlus: (autocompletePlus) ->
  this.subscriptions.add autocompletePlus.observeProviders (provider) ->
    if autocompletePlus.getDefaultProvider() is provider
      autocompletePlus.getDefaultProvider().addTextEditorSelector?('.find-container atom-text-editor') 
```

This would also prevent the details of the autocomplete-plus' implementation to leak into find-and-replace, because we wouldn't rely on the `SymbolProvider` magic constant anymore.

What do you think?